### PR TITLE
CPM-632: Implement empty/not empty filter on attribute and use it on product list REST API

### PR DESCRIPTION
**Pull request by Balamung**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/ListProducts/SuccessListProductEndToEnd.php [hunk 1]
```diff
@@ -939,6 +944,30 @@ public function testAccessDeniedWhenRetrievingProductsWithoutTheAcl()
...
+        $this->createProductWithoutIdentifier([
+            new SetTextValue('a_text', null, null, 'My product without identifier')
+        ]);
+
+        $client = $this->createAuthenticatedClient();
+        $client->request('GET', 'api/rest/v1/products?with_count=true&limit=10&pagination_type=search_after');
+        $result = json_decode($client->getResponse()->getContent(), true);
+        if (!isset($result['_embedded'])) {
+            Assert::fail($result);
+        }
```
Comment (mmetayer) [inline tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/ListProducts/SuccessListProductEndToEnd.php:960]:
```suggestion
        Assert::arrayHasKey('_embedded', $result);
```

File: src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Filter/Field/IdentifierFilter.php [hunk 2]
```diff
@@ -156,6 +159,21 @@ protected function applyFilter($field, $operator, $value)
...
+                ];
+
+                $this->searchQueryBuilder->addMustNot($clause);
+                break;
+
+            case Operators::IS_NOT_EMPTY:
+                $clause = [
+                    'exists' => ['field' => $field],
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
```
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Filter/Field/IdentifierFilter.php:174]: can you add an integration test please ?

**Review by mmetayer — APPROVED** [review]
GTM once integration tests are added

File: tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/ListProducts/SuccessListProductEndToEnd.php [hunk 3]
```diff
@@ -939,6 +944,30 @@ public function testAccessDeniedWhenRetrievingProductsWithoutTheAcl()
...
+
+        Assert::assertArrayNotHasKey('next', $result['_links']);
+
+        foreach ($result['_embedded']['items'] as $index => $product) {
+            if (isset($product['values']['a_text'])) {
+                Assert::assertNotEquals('My product without identifier', $product['values']['a_text'][0]['data']);
```
Comment (cemorin) [inline tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/ListProducts/SuccessListProductEndToEnd.php:966]: Replace with a check for uuid once frederics PR is merged

**Review by cemorin — APPROVED** [review]

File: tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/ListProducts/SuccessListProductEndToEnd.php (hunk 3 above)
Comment (Balamung) [reply tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/ListProducts/SuccessListProductEndToEnd.php:966]: Added comment and card to do that once the required PR is merged

**Pull request merged**